synapse/storage/databases/main/room.py

```diff
@@ -1370,9 +1370,10 @@ def _background_populate_rooms_creator_column_txn(txn: LoggingTransaction):
             """
 
             txn.execute(sql, (last_room_id, batch_size))
+            room_id_to_create_event_results = txn.fetchall()
 
             new_last_room_id = ""
-            for room_id, event_json in txn:
```

Comment:
Maybe I was also running into something similar to https://github.com/matrix-org/synapse/pull/10734

> We were originally iterating over `txn`, which can work, but as `txn` was being used to run a
simple_delete operation inside the loop, it was actually being redefined mid-loop.
>
> https://github.com/matrix-org/synapse/pull/10734

And in this case, we're iterating over `txn` and doing a `simple_update_txn` 🤔